### PR TITLE
Add advance search to roles listing

### PR DESCRIPTION
**Pull request by DimalChandrasiri**

### CODE REVIEW

File: apps/admin-portal/src/components/users/roles/roles-search.tsx [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+                                    value={ t("common:resetFilters").toString() }
+                                />
+                            </Form.Group>
+                        </Forms>
+                    </Grid.Column>
+                </Grid.Row>
+            </Grid>
+        </AdvancedSearch>
+    )
+}
```
Comment (inthirakumaaran) [inline apps/admin-portal/src/components/users/roles/roles-search.tsx]: Add a new line

Comment (DimalChandrasiri) [reply apps/admin-portal/src/components/users/roles/roles-search.tsx]: Fixed with commit 96511c4

File: apps/admin-portal/src/api/roles.ts [hunk 2]
```diff
@@ -54,6 +54,29 @@ export const getGroupsList = (domain: string): Promise<any> => {
...
+        headers: {
+            "Access-Control-Allow-Origin": GlobalConfig.clientHost,
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.POST,
+        url: ServiceResourcesEndpoint.groups + "/.search"
+    };
+
+    return httpClient.request(requestConfig).then((response) => {
+        return Promise.resolve(response);
```
Comment (brionmario) [inline apps/admin-portal/src/api/roles.ts]: Maybe we can cast the response to `RoleListInterface`.

```tsx
 response.data as RoleListInterface
```

**Return type**

```
Promise<RoleListInterface>
```

Comment (DimalChandrasiri) [reply apps/admin-portal/src/api/roles.ts]: Fixed with commit 22dbeb5

**Pull request merged**